analysis/reference-file-comparisons/compare_fasta_gtf.Rmd

```diff
@@ -127,58 +155,53 @@ genome_gtf <- genome_gtf[genome_gtf$transcript_id %in%
                            names(ensembl_genome)]
 
 # find the features that are in the eisaR gtf and also in the ensembl gtf
-shared_gtf <- txome_gtf[txome_gtf$transcript_id %in% 
-                          genome_gtf$transcript_id]
+shared_gtf <- genome_gtf[genome_gtf %in% txome_gtf]
 
 # get the features only found in the eisaR txome
-txome_only_gtf <- txome_gtf[!txome_gtf$transcript_id %in% 
-                              genome_gtf$transcript_id]
+txome_only_gtf <- txome_gtf[!txome_gtf %in% genome_gtf]
+
+# get the features only found in the ensembl gtf
+genome_only_gtf <- genome_gtf[!genome_gtf %in% txome_gtf]
```

Comment:
Why did you switch here to the full entries, rather than `transcript_id`? Does this make a difference in the counts?

Reply:
I was noticing that I was missing some of the overlap if I just did the transcript ID's and that it did make a difference in the counts, so it seemed more accurate to look at the entire entry rather than just the transcript ID.

Reply:
I guess I am not clear on what the comparison actually is then; comparing objects can have unexpected behavior, and I wonder if this might be inflating counts. 

A secondary question, which I meant to ask you but hadn't yet, is how many genes are actually affected? Are we losing individual transcripts from genes, or entire genes?

Reply:
So it looks like looking specifically at the gene level, there are no genes lost between the ensembl version and eisaR version. However, there are some genes that are gained in the eisaR version, not sure that gaining uncategorized genes is all that relevant though. The transcript level seems to be where the discrepancies are, so the question is do we care about that or not?

Reply:
I think the ultimate test is going to be to compare mapping to an index generated from the eisaR spliced fasta to the one generated from the ensembl reference fasta. If they are identical, then we don't care. If they differ...

Reply:
Very good point. I'm going to add in the workflows for generating the pre mRNA index files and then we can go from there in terms of comparing the eisaR spliced fasta to the ensembl one.